pkg/object/object.go

```diff
@@ -50,16 +71,129 @@ type OpenCompose struct {
 	Volumes  []Volume
 }
 
+func (s *Service) EmptyDirVolumeExists(name string) bool {
+
+	for _, emptyDirVolume := range s.EmptyDirVolumes {
+		if name == emptyDirVolume.Name {
+			return true
+		}
+	}
+	return false
+}
+
+func (o *OpenCompose) VolumeExists(name string) bool {
+	for _, volume := range o.Volumes {
+		if name == volume.Name {
+			return true
+		}
+	}
+	return false
+}
+
+// Documentation about the valid identifiers can be found at
+// https://github.com/kubernetes/community/blob/master/contributors/design-proposals/identifiers.md
+func isValidName(name string) error {
+	if errs := validation.IsDNS1123Subdomain(name); len(errs) != 0 {
+		return fmt.Errorf("%s", strings.Join(errs, "\n"))
+	}
+	return nil
+}
+
+func validatePath(path string) error {
+	// TODO: implement me
+	return nil
+}
+
+func (c *Container) Validate() error {
+
+	// validate image name
+	// TODO: implement me
+	// validate Environment
+	// TODO: implement me
+	// validate Ports
+	// TODO: implement me
+
+	// validate Mounts
+	for _, mount := range c.Mounts {
+		// validate volumeName
+		if err := isValidName(mount.VolumeName); err != nil {
+			return err
+		}
+
+		// validate mountPath
+		if err := validatePath(mount.MountPath); err != nil {
+			return err
+		}
+
+		// validate volumeSubPath
+		if err := validatePath(mount.VolumeSubPath); err != nil {
```

Comment:
What are rules for valid `VoumeSubPath`? Are there any that we can check?
If there are some we can create `validateSubPath` if there is nothing that we can check right now, we can skip sub path validation for now